### PR TITLE
refactor country code search

### PR DESCRIPTION
**Pull request by michaelkirk**
rebased https://github.com/WhisperSystems/SignalServiceKit/pull/78 (thanks @guillett!) and moved some remaining related tests from Signal-iOS.

closes #78 

corresponding ios PR: https://github.com/WhisperSystems/Signal-iOS/pull/1867/files

PTAL @charlesmchen

### CODE REVIEW

File: tests/Contacts/PhoneNumberTest.m [hunk 1]
```diff
@@ -15,6 +11,13 @@ @interface PhoneNumberTest : XCTestCase
 
 @implementation PhoneNumberTest
 
+-(void)testE164 {
+    XCTAssertEqualObjects(@"+19025555555", [[PhoneNumber tryParsePhoneNumberFromText:@"+1 (902) 555-5555" fromRegion:@"US"] toE164]);
+    XCTAssertEqualObjects(@"+19025555555", [[PhoneNumber tryParsePhoneNumberFromText:@"1 (902) 555-5555" fromRegion:@"US"] toE164]);
+    XCTAssertEqualObjects(@"+19025555555", [[PhoneNumber tryParsePhoneNumberFromText:@"1-902-555-5555" fromRegion:@"US"] toE164]);
+    XCTAssertEqualObjects(@"+19025555555", [[PhoneNumber tryParsePhoneNumberFromText:@"1-902-５５５-5555" fromRegion:@"US"] toE164]);
```
Comment (charlesmchen) [inline tests/Contacts/PhoneNumberTest.m:18]: Would it also be worthwhile to check other input:

* Non-US calling codes.
* Phone numbers missing a calling code.
* Phone numbers with a calling code but without a +.
* Empty input.
* Incomplete (short) input.
* Overlong input.
* Bad input.

File: tests/Contacts/PhoneNumberUtilTest.m [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    XCTAssertTrue([PhoneNumberUtil name:@"big dave" matchesQuery:@"      big       "]);
+    XCTAssertTrue([PhoneNumberUtil name:@"big dave" matchesQuery:@"dav"]);
+    XCTAssertTrue([PhoneNumberUtil name:@"big dave" matchesQuery:@"bi dav"]);
+    XCTAssertTrue([PhoneNumberUtil name:@"big dave"
+                           matchesQuery:@"big big big big big big big big big big dave dave dave dave dave"]);
+
+    XCTAssertFalse([PhoneNumberUtil name:@"big dave" matchesQuery:@"ave"]);
+    XCTAssertFalse([PhoneNumberUtil name:@"big dave" matchesQuery:@"dare"]);
+    XCTAssertFalse([PhoneNumberUtil name:@"big dave" matchesQuery:@"mike"]);
+    XCTAssertFalse([PhoneNumberUtil name:@"big dave" matchesQuery:@"mike"]);
```
Comment (charlesmchen) [inline tests/Contacts/PhoneNumberUtilTest.m:30]: duplicate?

**Review by charlesmchen — APPROVED** [review]
LGTM

File: tests/Contacts/PhoneNumberUtilTest.m [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+@interface PhoneNumberUtilTest : XCTestCase
+
+@end
+
+@implementation PhoneNumberUtilTest
+
+- (void)testQueryMatching
+{
+    XCTAssertTrue([PhoneNumberUtil name:@"big dave" matchesQuery:@"big dave"]);
```
Comment (charlesmchen) [inline tests/Contacts/PhoneNumberUtilTest.m:16]: XCTAssertTrue([PhoneNumberUtil name:@"big dave" matchesQuery:@"big dave dave"]);
XCTAssertTrue([PhoneNumberUtil name:@"big dave dave" matchesQuery:@"big dave"]);

Comment (michaelkirk) [reply tests/Contacts/PhoneNumberUtilTest.m:16]: added

File: tests/Contacts/PhoneNumberUtilTest.m (hunk 2 above)
Comment (michaelkirk) [reply tests/Contacts/PhoneNumberUtilTest.m:30]: removed

**Comment by michaelkirk** [comment]
In https://github.com/WhisperSystems/SignalServiceKit/commit/ca5bcaf3c2525555cfe140a78b1850632db1d440

**Pull request closed without merging**